src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -1028,6 +1018,83 @@ private JButton getValidateButton() {
 		});
 		return validateButton;
 	}
+	private JButton getConfirmButton() {
+		confirmButton = new JButton(MessageBundle.getMessage("angal.inventory.confirm.btn"));
+		confirmButton.setMnemonic(MessageBundle.getMnemonic("angal.inventory.confirm.btn.key"));
+		if (inventory == null) {
+			confirmButton.setEnabled(false);
+		}
+		confirmButton.addActionListener(actionEvent -> {
+			if (inventory == null) {
+				MessageDialog.error(null, "angal.inventory.inventorymustsavebeforevalidation.msg");
+				return;
+			}
+			List<MedicalInventoryRow> invRowWithoutLot = inventoryRowSearchList.stream().filter(invRow -> invRow.getLot() == null).collect(Collectors.toList());
+			if (invRowWithoutLot.size() > 0) {
+				MessageDialog.error(null, "angal.inventory.allinventoryrowshouldhavelotbeforevalidation.msg");
+				return;
+			}
+			int confirm = MessageDialog.yesNo(null, "angal.inventory.doyoureallywanttoconfirmthisinventory.msg");
+			if (confirm == JOptionPane.YES_OPTION) {
+				String dischargeCode = inventory.getDischargeType();
+				String chargeCode = inventory.getChargeType();
+				Integer supplierId = inventory.getSupplier();
+				String wardCode = inventory.getDestination();
+				String errorMessage = this.checkParamsValues(chargeCode, dischargeCode, supplierId, wardCode);
+				if (errorMessage != null) {
+					MessageDialog.error(null, errorMessage);
+					return ;
+				}
+				// confirm inventory
+				try {
+					medicalInventoryManager.confirmMedicalInventoryRow(inventory, inventoryRowSearchList);					
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+					try {
+						jTableInventoryRow.setModel(new InventoryRowModel());
+						adjustWidth();
+					} catch (OHServiceException e1) {
+						OHServiceExceptionUtil.showMessages(e1);
+					}
+					return;
+				}
+				String status = InventoryStatus.done.toString();
+				inventory.setStatus(status);
+				try {
+					inventory = medicalInventoryManager.updateMedicalInventory(inventory);
+					if (inventory != null) {
+						MessageDialog.info(null, "angal.inventory.confirm.success.msg");
+						fireInventoryUpdated();
+						closeButton.doClick();
+					} else {
+						MessageDialog.info(null, "angal.inventory.confirm.error.msg");
+						return;
+					}
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+					return;
+				}
```

Comment:
Noticed now while testing: should the inventory been moved automatically to "done" status upon a successfull confirmation? Otherwise will be always the GUI (or the UI) to do such operation... wdyt?
If so, also the "validated" status should be automatically set upon validation success.

Reply:
I think it good to automatically do it after confirmation

Reply:
ok then, waiting for the change

Reply:
should this change after automatic validation and confirmation statuses in CORE?

Reply:
yes

Reply:
ok, so waiting the change